satellite/metainfo/metainfo.go

```diff
@@ -475,18 +479,38 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 	defer mon.Task()(&ctx)(&err)
 
 	if pointer.Type == pb.Pointer_REMOTE {
-		var remotePieces []*pb.RemotePiece
 		remote := pointer.Remote
+
+		nodeIDList := storj.NodeIDList{}
+		for _, piece := range remote.RemotePieces {
+			nodeIDList = append(nodeIDList, piece.NodeId)
+		}
+		peerIDList, err := endpoint.peerIdentities.BatchGet(ctx, nodeIDList)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+		peerIDMap := make(map[storj.NodeID]*identity.PeerIdentity, len(peerIDList))
+		for _, peerID := range peerIDList {
+			peerIDMap[peerID.ID] = peerID
+		}
+
+		var remotePieces []*pb.RemotePiece
 		allSizesValid := true
 		lastPieceSize := int64(0)
 		for _, piece := range remote.RemotePieces {
 
-			// TODO enable piece hash signature verification
+			// Verify storagenode signature on piecehash
+			peerID := peerIDMap[piece.NodeId]
+			if peerID == nil {
+				endpoint.log.Warn("Peer ID is nil for node", zap.String("nodeID", piece.NodeId.String()))
```

Comment:
It looks like this could only happen when we don't have the chain stored yet. Perhaps it would help for the error message to indicate this, so the future log reader doesn't have to go back and search. "Identity chain unknown for node" maybe?

Reply:
Done